sky/data/sky_csync.py

```diff
@@ -0,0 +1,337 @@
+    while True:
+        start_time = time.time()
+        run_sync(full_src, storetype, destination, num_threads,
+                 interval_seconds, delete, no_follow_symlinks, csync_pid)
+        end_time = time.time()
+        # Given the interval_seconds and the time elapsed during the sync
+        # operation, we compute remaining time to wait before the next
+        # sync operation.
+        elapsed_time = int(end_time - start_time)
+        remaining_interval = max(0, interval_seconds - elapsed_time)
+        # sync_pid column is set to 0 when sync is not running
+        time.sleep(remaining_interval)
+
+
+@main.command()
+@click.argument('paths', nargs=-1, required=False, type=str)
+@click.option('--all',
+              '-a',
+              default=False,
+              is_flag=True,
+              required=False,
+              help='Terminates all CSYNC processes.')
+def terminate(paths: List[str], all: bool = False) -> None:  # pylint: disable=redefined-builtin
+    """Terminates all the CSYNC daemon running after checking if all the
+    sync process has completed.
+
+    Args:
+        paths (List[str]): list of CSYNC-mounted paths
+        all (bool): determine either or not to unmount every CSYNC-mounted
+            paths
+
+    Raises:
+        click.UsageError: when the paths are not specified
+    """
+    if not paths and not all:
+        raise click.UsageError('Please provide the CSYNC-mounted path to '
+                               'terminate the CSYNC process.')
+    _terminate(paths, all)
+
+
+def _terminate(paths: List[str], all: bool = False) -> None:  # pylint: disable=redefined-builtin
+    """Terminates all the CSYNC daemon running after checking if all the
+    sync process has completed.
+    """
+    # TODO(Doyoung): Currently, this terminates all the CSYNC daemon by
+    # default. Make an option of --all to terminate all and make the default
+    # behavior to take a source name to terminate only one daemon.
+    # Call the function to terminate the csync processes here
+    if all:
+        csync_pid_set = set(_get_all_running_csync_pid())
+    else:
+        csync_pid_set = set()
+        for path in paths:
+            full_path = os.path.abspath(os.path.expanduser(path))
+            csync_pid_set.add(_get_csync_pid_from_source_path(full_path))
+    while True:
+        if not csync_pid_set:
+            break
+        sync_running_csync_set = set()
+        for csync_pid in csync_pid_set:
+            # sync_pid is set to -1 when sync is not running
+            if _get_running_csync_sync_pid(csync_pid) != -1:
+                sync_running_csync_set.add(csync_pid)
+        remove_process_set = csync_pid_set.difference(sync_running_csync_set)
+        for csync_pid in remove_process_set:
+            try:
+                psutil.Process(int(csync_pid)).terminate()
+            except psutil.NoSuchProcess as e:
+                if 'process no longer exists' in str(e):
```

Comment:
What if `'process no longer exists' not in str(e)`?

Reply:
Good point. Just removed the `if` condition. Catching `NoSuchProcess` would be an enough indicator to clean the db.